User/src/main/java/com/example/user/service/UserService.java

```diff
@@ -0,0 +1,70 @@
+package com.example.user.service;
+
+import com.example.user.Controller.UserDtoMapper;
+import com.example.user.Repository.UserRepository;
+import com.example.user.exceptions.ErrorSubmissionException;
+import com.example.user.model.CredentialsDto;
+import com.example.user.model.RegistrationDto;
+import com.example.user.model.UserDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.bcrypt.BCrypt;
+import org.springframework.stereotype.Service;
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepository;
+    private final UserDtoMapper dtoMapper;
+    public UserDto getUserById(Long id) {
+        return userRepository.findById(id).map(dtoMapper::mapToDto)
+                .orElseThrow(()->new ErrorSubmissionException("Not found user id"));
+    }
+
+    public UserDto changeEmail(long id, String passw, String newEmail) {
+        return userRepository.findById(id)
+                .filter(user->BCrypt.checkpw(passw, user.getPassw())).map(user->{
+                    user.setEmail(newEmail);
+                    userRepository.save(user);
+                    return dtoMapper.mapToDto(user);
+                        }).orElseThrow(()->new ErrorSubmissionException("User id not found"));
+    }
+
+    public List<UserDto> getUserBySearchValue(String searchValue) {
+        return dtoMapper.mapToListDto(userRepository.findUserByEmailOrFirstNameOrLastName(
+                searchValue, searchValue, searchValue).stream().toList());
```

Comment:
Nie wiem po co masz tutaj ".stream().tolist()". Jeśli wolisz zwracać listę, to zmień metodę w repo na List bo to niepotrzebna operacja